aptos-move/framework/aptos-framework/sources/account/account.move

```diff
@@ -424,6 +443,185 @@ module aptos_framework::account {
+        // Check that the public key is a keyless public key by checking the scheme
+        // encoded in the first byte of the single key public key
+        let public_key_type = keyless_public_key[0];
+        assert!(public_key_type == 3 || public_key_type == 4, error::invalid_argument(ENOT_A_KEYLESS_PUBLIC_KEY));
+        let addr = signer::address_of(account);
+        let account_resource = &mut Account[addr];
+
+        // Check that the provided public key is the current public key of the account by comparing
+        // its authentication key to the account's authentication key
+        let auth_key = get_authentication_key_from_scheme_and_public_key_bytes(SINGLE_KEY_SCHEME, keyless_public_key);
+        assert!(
+            auth_key == account_resource.authentication_key,
+            error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY)
+        );
+
+        // Check that the keyless public key is the original public key of the account by comparing
+        // its authentication key to the account address.
+        assert!(
+            bcs::to_bytes(&addr) == auth_key,
+            error::invalid_argument(ENOT_THE_ORIGINAL_PUBLIC_KEY)
+        );
+        let curr_auth_key_as_address = from_bcs::to_address(account_resource.authentication_key);
+        let challenge = RotationProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            originator: addr,
+            current_auth_key: curr_auth_key_as_address,
+            new_public_key: backup_key,
+        };
+
+        // Assert the challenges signed by the provided backup key is valid
+        assert_valid_rotation_proof_signature_and_get_auth_key(
+            ED25519_SCHEME,
+            backup_key,
+            backup_key_proof,
+            &challenge
+        );
+
+        // Get the backup key as a single key
+        let backup_key_ed25519 = ed25519::new_unvalidated_public_key_from_bytes(backup_key);
+        let backup_key_as_single_key = single_key::from_ed25519_public_key_unvalidated(&backup_key_ed25519);
+
+        // Construct the multi key public key which should be the current public key of the account
+        let new_public_key_bytes = vector[];
+        new_public_key_bytes.push_back(0x02); // Number of keys in the multi key
+        new_public_key_bytes.append(keyless_public_key);  // Add the current key which is the keyless public key
+        new_public_key_bytes.append(single_key::unvalidated_public_key_to_bytes(&backup_key_as_single_key));
+        new_public_key_bytes.push_back(0x01); // Signatures required
+
+        let new_auth_key = get_authentication_key_from_scheme_and_public_key_bytes(MULTI_KEY_SCHEME, new_public_key_bytes);
+
+        // Rotate the authentication key to the new multi key public key
+        rotate_authentication_key_call(account, new_auth_key);
+
+        event::emit(KeyRotationToMultiPublicKey {
+            account: addr,
+            verified_public_key_bit_map: vector[0xC0, 0x00, 0x00, 0x00],
```

Comment:
What is this hardcoded vector[0xC0, 0x00, 0x00, 0x00]

Reply:
This represents the 'signature bitmap' which indicates what public keys of the multikey are owned.  It is four bytes wide.



vector[0xC0, 0x00, 0x00, 0x00] represents 0b1100000000000000000000000000000

which indicates public keys at index 0 and 1 are verified.

0 (the keyless key) - because it is the original key
1 (backup) - because of the proof verification

Reply:
Add to the comment please